chalicelib/lib/db_v2.py

```diff
@@ -0,0 +1,229 @@
+
+
+def get_event_by_user_id_and_date(user_id, event_date):
+    """
+    :param user_id:
+    :param event_date:
+    :return: event
+    """
+    try:
+        e = EventTable.get(user_id, event_date)
+        return e.attribute_values
+    except EventTable.DoesNotExist as e:
+        return json.dumps({})
+
+
+def delete_event_by_user_id_and_date(user_id, event_date):
+    """
+    :param user_id:
+    :param event_date:
+    :return: method, user_id, count, status
+    """
+    count = 0
+    events = EventTable.scan((EventTable.user_id == user_id) & (EventTable.event_date == event_date))
+    for event in events:
+        event.delete()
+        count += 1
+    return json.dumps({"method": f"delete", "user_id": f"{user_id}", "count": f"{count}",
+                       "date": f"{event_date}", "status": "ok"})
+
+
+def list_locks_by_user_id(user_id):
+    """
+    :param user_id:
+    :return: list of locks for user
+    """
+    locks = []
+    for lock in LockTable.scan(LockTable.user_id == user_id):
+        locks.append(lock.attribute_values)
+    return json.dumps(locks)
+
+
+def delete_all_locks_by_user_id(user_id):
+    """
+    :param user_id:
+    :return: method, user_id, count, status
+    """
+    count = 0
+    for lock in LockTable.scan(LockTable.user_id == user_id):
+        lock.delete()
+        count += 1
+    return json.dumps({"method": f"delete", "user_id": f"{user_id}", "count": f"{count}", "status": "ok"})
+
+
+def get_lock_by_user_id_and_date(user_id, event_date):
+    """
+    :param user_id:
+    :param event_date:
+    :return: lock for user
+    """
+    locks = LockTable.scan(LockTable.user_id == user_id, LockTable.event_date == event_date)
+    for lock in locks:
+        lock.delete()
+    return json.dumps({"Method": f"DELETE", "user_id": f"{user_id}", "date": f"{event_date}", "Status": "OK"})
+
+
+def delete_lock_by_user_id_and_date(user_id, event_date):
+    """
+    :param user_id:
+    :param event_date:
+    :return: status
+    """
+    locks = LockTable.scan((LockTable.user_id == user_id) & (LockTable.event_date == event_date))
+    for lock in locks:
+        lock.delete()
+    return json.dumps({"Method": f"DELETE", "user_id": f"{user_id}", "date": f"{event_date}", "Status": "OK"})
+
+
+def list_all_events():
+    """
+    :return: list of all events
+    """
+    events = []
+    for event in EventTable.scan():
+        events.append(event.attribute_values)
+    return json.dumps(events)
+
+
+def list_all_events_by_date(event_date):
+    """
+    :param event_date:
+    :return: list of all events for date
+    """
+    events = []
+    for event in EventTable.scan(EventTable.event_date == event_date):
+        events.append(event.attribute_values)
+    return json.dumps(events)
+
+
+def create_event_v2(events):
+    """
+    :param events:
+    :return: status
+    """
+    if isinstance(events, str):
+        events = ast.literal_eval(events)
```

Comment:
Can you explain why `events` might be a string and what this `ast.literal_eval(events)` will do?

Reply:
Hmm, dont know

https://github.com/codelabsab/timereport-api/commit/52be417fd263e2ab826d2a4be857915e4813ae6b

Kami introduced it some time ago